sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/OperatorStateMetadataSuite.scala

```diff
@@ -32,30 +32,37 @@ class OperatorStateMetadataSuite extends StreamTest with SharedSparkSession {
 
   private lazy val hadoopConf = spark.sessionState.newHadoopConf()
 
+  private var checkpointDir = Utils.createTempDir()
```

Comment:
Our best practice is to use withTempDir per each test. Could we please follow the pattern and remove this and beforeEach?

Reply:
Refactored.